app/frontend/terrier/parts/select-field-part.ts

```diff
@@ -0,0 +1,214 @@
+            this.clear()
+        })
+    }
+    get autoClose(): boolean {
+        return true
+    }
+
+    get parentClasses(): Array<string> {
+        return ['tt-select-options-dropdown', ...super.parentClasses]
+    }
+
+    renderContent(parent: PartTag) {
+        for (const opt of this.state.options) {
+            if ('group' in opt) {
+                parent.div('.tt-select-options-group-header', { text: opt.group } )
+                for (const groupOpt of opt.options) {
+                    this.renderOption(parent, groupOpt)
+                }
+            } else {
+                this.renderOption(parent, opt)
+            }
+        }
+    }
+
+    renderOption(parent: PartTag, opt: SelectOption) {
+        parent.div({ text: opt.title }, option => {
+            if (this.state.selected_option.value == opt.value) option.class('selected')
+        }).emitClick(this._clickedOptionKey, { selected_option: opt })
+    }
+
+    update(elem: HTMLElement) {
+        const dropdown = elem.querySelector('.tt-dropdown-content') as HTMLElement
+        if (dropdown) {
+            const selectedElement = dropdown.querySelector('.selected') as HTMLElement
+            if (this.anchorTarget && selectedElement) {
+                this.anchorSelectDropdown(dropdown, this.anchorTarget, selectedElement)
+            }
+        }
+    }
+
+    /**
+     * Anchors a select dropdown on top of a select field. Uses the currently selected child option to determine positioning.
+     * Assumes that all child elements have the same height.
+     * @param dropdown the element to reposition
+     * @param selectField the anchor select field
+     * @param selectedElement the selected option that will be positioned directly above the select field
+     */
+    anchorSelectDropdown(dropdown: HTMLElement, selectField: HTMLElement, selectedElement: HTMLElement) {
+        const dropdownSize = Overlays.getElementSize(dropdown)
+        const selectOptionSize = Overlays.getElementSize(selectedElement)
+
+        const selectedOptionIndex = Array.prototype.indexOf.call(dropdown.childNodes, selectedElement)
+
+        const anchorRect = selectField.getBoundingClientRect()
+        const win = Overlays.getWindowSize()
+
+        const result: AnchorResult = {
+            top: anchorRect.y - (selectOptionSize.height * selectedOptionIndex),
+            left: anchorRect.x,
+            valid: true
+        }
+
+        Overlays.clampAnchorResult(result, dropdownSize, win)
+
+        if (dropdownSize.width < anchorRect.width) {
+            result.width = anchorRect.width
+        }
+
+        // how far down the select options list we need to scroll
+        let scrollAmount = (selectOptionSize.height * (selectedOptionIndex))
+        const hasOptGroups = this.state.options.find((o) => 'group' in o)
+        if (hasOptGroups) {
+            scrollAmount -= selectOptionSize.height
+        }
```

Comment:
Can't we just use `selectedElement.offsetTop` here? that should give us the position of the selected element relative to thedropdown.

Reply:
😎